mystery_word.py

```diff
@@ -0,0 +1,115 @@
+import random
+
+
+class MysteryWord:
+
+    default_dictionary = "/usr/share/dict/words"
+
+    def __init__(self, file=default_dictionary, difficulty=0):
+        """Initializes and instance of the game.  Loads the dictionary file,
+        asks user for difficulty, selects a word from the dictionary based
+        on difficulty level, sets number of guesses, splits the word into
+        a list of tuples containing the letter plus "FALSE" value indicating
+        that the letter has not been guessed by the player."""
+
+        word_list = self.load_dictionary(file)
+        self.word = self.choose_word(word_list, difficulty).upper()
+        self.word_length = len(self.word)
+        self.prepared_word = self.prepare_word_for_game()
+        self.remaining_guesses = 8
+        self.guessed_letters = []
+
+    def __str__(self):
+        word_status_string = ""
+        for letter in self.prepared_word:
+            if letter[1]:
+                word_status_string += " {}".format(letter[0])
+            else:
+                word_status_string += " _"
+
+        return word_status_string
+
+    def check_guess(self, player_guess):
+        """ Compares the player's guess to the word to guess."""
+
+        player_guess = player_guess.upper()
+
+        if player_guess == "QUIT":
+            exit()
+        elif not player_guess.isalpha() or len(player_guess) > 1:
+            raise ValueError
+        elif player_guess in self.guessed_letters:
+            raise LookupError
+        elif player_guess in self.word:
+            for letter in self.prepared_word:
+                if player_guess == letter[0]:
+                    letter[1] = True
+            self.guessed_letters.append(player_guess)
+            return True
+        else:
+            self.guessed_letters.append(player_guess)
+            self.remaining_guesses -= 1
+            return False
+
+    def check_win_lose(self):
+        """ Checks to see if the player has won or lost the game """
+
+        if not self.remaining_guesses:
+            return False
+        elif False not in [element[1] for element in self.prepared_word]:
+            return True
+
+    def continue_game(self):
+        """Checks if an end-game state has been reached.  Returns True if the
+        the game should continue, False if the game has ended."""
+
+        if not self.remaining_guesses or False not in [element[1] for element
+                                                       in self.prepared_word]:
+            return False
+        else:
+            return True
+
```

Comment:
Whenever you have a statement like:

```py
if predicate:
  return True
else:
  return False
```

or its opposite, you can simplify it down. For this, you could have:

```py
return self.remaining_guesses and False in [element[1] for element in self.prepared_word]
# or
return self.remaining_guesses and not all(element[1] for element in self.prepared_word)
```

I think I have [DeMorgan's Law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) right on that one.